### PR TITLE
Implement unbound_exporter

### PR DESCRIPTION
**Pull request by bastelfreak**

### CODE REVIEW

File: manifests/unbound_exporter.pp [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+#  Extra options added to the startup command
+# @param group
+#  Group under which the binary is running
+# @param init_style
+#  Service startup scripts style (e.g. rc, upstart or systemd)
+# @param install_method
+#  Installation method: url or package (only url is supported currently)
+# @param manage_group
+#  Whether to create a group for or rely on external code for that
+# @param manage_service
```
Comment (alexjfisher) [inline manifests/unbound_exporter.pp:26]: No need to specify defaults when puppet-strings will already work these out.

File: manifests/unbound_exporter.pp [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+# @param purge_config_dir
+#  Purge config files no longer generated by Puppet
+# @param restart_on_change
+#  Should puppet restart the service on configuration change? (default true)
+# @param service_enable
+#  Whether to enable the service from puppet (default true)
+# @param service_ensure
+#  State ensured for the service (default 'running')
+# @param service_name
+#  Name of the bird exporter service (default 'bird_exporter')
```
Comment (alexjfisher) [inline manifests/unbound_exporter.pp]: The what exporter??

File: manifests/unbound_exporter.pp [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+# @param manage_service
+#  Should puppet manage the service? (default true)
+# @param manage_user
+#  Whether to create user or rely on external code for that
+# @param os
+#  Operating system (linux is the only one supported)
+# @param package_ensure
+#  If package, then use this for package ensure default 'latest'
+# @param package_name
+#  The binary package name - not available yet
```
Comment (alexjfisher) [inline manifests/unbound_exporter.pp:35]: Not available?

**Review by alexjfisher — DISMISSED** [review]

File: manifests/unbound_exporter.pp (hunk 2 above)
Comment (bastelfreak) [reply manifests/unbound_exporter.pp]: oh my sed failed me here. i copied the bird exporter

File: manifests/unbound_exporter.pp (hunk 3 above)
Comment (bastelfreak) [reply manifests/unbound_exporter.pp:35]: see also the summary. the exporter is currently only released as go source code and you need to compile it yourself (hence the install_method = `none`). I am tracking this upstream and hope binaries will be released soon.

File: manifests/unbound_exporter.pp (hunk 1 above)
Comment (bastelfreak) [reply manifests/unbound_exporter.pp:26]: removed some defaults

File: manifests/unbound_exporter.pp (hunk 2 above)
Comment (bastelfreak) [reply manifests/unbound_exporter.pp]: fixed it

File: manifests/unbound_exporter.pp [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+# @param extra_groups
+#  Extra groups to add the binary user to
+# @param extra_options
+#  Extra options added to the startup command
+# @param group
+#  Group under which the binary is running
+# @param init_style
+#  Service startup scripts style (e.g. rc, upstart or systemd)
+# @param install_method
+#  Installation method: url or package (only url is supported currently)
```
Comment (alexjfisher) [inline manifests/unbound_exporter.pp:23]: Is `url` supported?  Or it is if URL is to a local mirror where you've built the binary??
Maybe include an `@example Installing from your own package repository` or something based on how you're currently using this?

**Review by alexjfisher — APPROVED** [review]
I think this is probably ok.  Maybe just add an example for added clarity on how a user might use this.

File: manifests/unbound_exporter.pp (hunk 4 above)
Comment (bastelfreak) [reply manifests/unbound_exporter.pp:23]: at the moment I compiled the binary by myself and copied it into /usr/local/bin/unbound_exporter. I was hoping that https://github.com/kumina/unbound_exporter/issues/21 will be solved soon. That would enable this class to download the exporter from there. maybe we should keep this PR open until they provide proper releases.

**Pull request merged**